Sources/PackageGraph/TraitConfiguration.swift

```diff
@@ -11,18 +11,96 @@
 //===----------------------------------------------------------------------===//
 
 /// The trait configuration.
-public struct TraitConfiguration: Codable, Hashable {
-    /// The traits to enable for the package.
-    package var enabledTraits: Set<String>?
-
-    /// Enables all traits of the package.
-    package var enableAllTraits: Bool
+public enum TraitConfiguration: Codable, Hashable {
+    case enableAllTraits
+    case noConfiguration
```

Comment:
question: what does `noConfiguration` represent?  how is it different than the `noEnabledTraits`?  is there an actual behavioural difference between `noConfiguration` and `noEnabledTraits`?

Reply:
There is, yes: I've since changed the nomenclature for these cases and will be pushing that up today, but essentially `noConfiguration` encapsulates the case where there is no configuration of traits passed in by the user and as such will default to using a package's default traits if there are any. If a user _does_ explicitly pass in a configuration with an empty set of traits enabled (so essentially, no traits enabled), this overrides the default traits behaviour and instead disables all traits which is what `.noEnabledTraits` accomplishes. :)